packages/components/icon/src/Icon.tsx

```diff
@@ -0,0 +1,147 @@
+import { css, cx } from 'emotion';
+import React, { forwardRef } from 'react';
+import tokens from '@contentful/f36-tokens';
+import { Box } from '@contentful/f36-core';
+import type {
+  CommonProps,
+  BoxProps,
+  ComponentVariant,
+  MarginProps,
+  PaddingProps,
+  PolymorphicComponent,
+  PolymorphicComponentProps,
+  PolymorphicComponentWithRef,
+  Simplify,
+} from '@contentful/f36-core';
+import type {
+  ComponentType,
+  ExoticComponent,
+  ReactElement,
+  SVGAttributes,
+} from 'react';
+
+const DEFAULT_TAG = 'svg';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type IconComponent = ExoticComponent<any> | ComponentType<any>;
+
+export type IconSize = 'large' | 'medium' | 'small' | 'tiny';
+
+export type IconVariant = Simplify<ComponentVariant | 'muted' | 'white'>;
+
+const sizes: { [key in IconSize]: { [key in 'height' | 'width']: string } } = {
+  large: {
+    height: '32px',
+    width: '32px',
+  },
+  medium: {
+    height: '24px',
+    width: '24px',
+  },
+  small: {
+    height: '18px',
+    width: '18px',
+  },
+  tiny: {
+    height: '16px',
+    width: '16px',
+  },
+};
+
+const fills: { [key in IconVariant]: string } = {
+  muted: tokens.colorTextLight,
+  negative: tokens.colorNegative,
+  positive: tokens.colorPositive,
+  primary: tokens.colorBlueBase,
+  secondary: tokens.colorContrastMid,
+  warning: tokens.colorWarning,
+  white: tokens.colorWhite,
+};
+
+type AsOrChildren =
+  | {
+      as: IconComponent;
+      children?: never;
+    }
+  | {
+      as?: never;
+      children: ReactElement | ReactElement[];
+    };
+
+type IconInternalProps = {
+  /**
+   * Determines the size of the icon
+   */
+  size?: IconSize;
+  /**
+   * Whether or not to trim the icon width, i.e. set `width` to `auto`
+   */
+  trimmed?: boolean;
```

Comment:
Wouldn't that introduce sizes inconsistencies?

Reply:
Sure, but trimmed icons already have that.

Reply:
Yes, I mean should we remove it then with the new sizes, and V4?

Reply:
I can't say, we should probably run it by design. I don't know where the trimmed icons are used or how much.